### ISSUE TITLE
New UI concept

### ISSUE THREAD

**Issue by vfsfitvnm** [vfsfitvnm/ViMusic#496]
As you all know, the version 0.5 brought a massive UI change. Some people are concerned about the ease of navigation, and I can't blame them (full discussion: #390).
I honestly love how every screen looks like, but perhaps it's more good looking than functional.
An aspect I don't like is the difficulty to tell _where_ you actually are: every screen resembles literally any other screen of the whole application.

The vertical navigation bar, also known as navigation rail, improved the UI, but harmed the usability (i.e. your finger has to move too much to change the content on the screen).
I took some spare time to address the issues, but as soon as I removed the vertical bar (_the root of evil_) the screens looked arid - as they were just a pile of stuff glued together.

I'm not sure how these things matter in a music player app, however I always try to push things further so I started thinking about another design.

All music players have the same layout: bottom navigation bar (or navigation rail + horizontal tabs) + bottom player (that can be expanded). That's it. Thanks, material design.

I already do not use a top/bottom navigation component and now I'd like to see if I can get rid of the bottom player, too.

The concept I had in mind is having an _immersive_ player screen as the main screen of the application. With a lot of gestures.

Here it is:

https://user-images.githubusercontent.com/46219656/197348666-c436b3e0-3bd2-4de5-9a9b-75d571ff0fe7.mp4

- Translucent theme - the playing song cover is the background of the whole app
- Swipe up/down to change song
- Swipe right to display the queue
- Swipe left to access the library (not yet implemented)
- Full screen lyrics (not yet implemented)

I'd like to know to hear what you think before investing a lot of time in this concept. It was fun to do :yum:

**Comment by brnwlshubh** [comment]
It really looks great

**Comment by sidsharths1245** [comment]
It looks excellent and it seems pretty easy and straightforward to use. In my personal opinion the player screen looks a bit empty, but i think once you implement the full screen lyrics it should look great.

One suggestion i have is that when you add the library option you can have two separate tabs. One called "Explore" as in the current quick picks and the other named "personal" or whatever for our added songs and playlists etc.

Another one i realised while typing is something that you can experiment with to see if it looks better or worse. Like you can have the album art as a square in the middle of the screen instead of being stretched over the entire screen.

**Comment by sidsharths1245** [comment]
> It looks excellent and it seems pretty easy and straightforward to use. In my personal opinion the player screen looks a bit empty, but i think once you implement the full screen lyrics it should look great.
> 
> One suggestion i have is that when you add the library option you can have two separate tabs. One called "Explore" as in the current quick picks and the other named "personal" or whatever for our added songs and playlists etc and all the elements (artists, playlists, songs) can be added inside these two sections.
> 
> Another one i realised while typing is something that you can experiment with to see if it looks better or worse. Like you can have the album art as a square in the middle of the screen instead of being stretched over the entire screen.
> 
> Edit: I absolutely love this concept. I just realised how unique it looks. Great idea man. It takes great effort to go back to the drawing board once you're at such stage in development with the current design. Keep up the great work.

One last thing i would suggest is that you could have other important buttons like, loop, add to playlist, start radio etc as marked in the picture below. It would make the screen not look so empty.
![Screenshot_20221022-224441](https://user-images.githubusercontent.com/64619769/197353669-bae7e61a-2e74-4eb4-9278-c740205dc15a.png)

**Comment by vfsfitvnm** [comment]
Thanks!
And yes, I think I will add two more buttons (or gestures) at least to set the loop mode and show the menu/lyrics/stats for nerds!

Unfortunately the thumbnail must be at full screen, the screen looks empty otherwise and the gestures becomes innatural

**Comment by vfsfitvnm** [comment]
@PassionateG1t 
I get your point about this type of layout. Not sure if I was a little mischievous when I thought of this, but the only thing being shared here is the vertical scroll only. There's nothing addicting because there's no reason to "keep scrolling" (but the apps you mentioned do). Would you scroll to the next song to stare at a still image for 15 seconds, hear like a twentieth of the song and then proceed to scroll again? Also, what forbids you to do the exact same thing with the current layout? Not all evil comes to harm!

**Comment by Fatcakes** [comment]
I think this a step in the wrong direction. The player page already looked great. Players look like they do for a reason. Ease of use and functionality. You're sacrifice both for imo a worse layout. If you're doing this for the lyrics then I guess I don't understand. The way you have implemented lyrics has been satisfactory for awhile now.

**Comment by arfanmrafeek** [comment]
Easy, Love the design. Waiting for finalized result.

**Comment by arfanmrafeek** [comment]
Before finial release, We need to have our hands on it experience

**Comment by NeeRaj-2401** [comment]
@vfsfitvnm I appreciate ur efforts to make it better day by day and try something unique. But honestly this tiktok inspired new layout doesn't suits with a music streaming app. Agreeing with @PassionateG1t

**Comment by scottiefalkon** [comment]
I can also say that I love the current UI.

**Comment by vfsfitvnm** [comment]
Needless to say this view can be just incorporated to the current UI (this is how it would look like when you expand the player). In such case:
- swipe left/right to go to previous/next song
- swipe down to collapse
- swipe up to show the queue
etc

**Comment by arfanmrafeek** [comment]
swipe down twice to get search. Maybe.

**Comment by NeeRaj-2401** [comment]
> Needless to say this view can be just incorporated to the current UI (this is how it would look like when you expand the player).

 
current player layout is amazing. i think there is no need to change it. 
the major concern is to replace left navigation rail.. and that is pretty much solved here https://github.com/vfsfitvnm/ViMusic/issues/390#issuecomment-1284139296 in my opinion

**Comment by scottiefalkon** [comment]
Actually. Looking again. If you're going to ditch the sidebar, x the full
screen swipe UI concept is really clean. I like I more than just adding
section after section to one screen. Too busy. Too much scrolling.

**Comment by NeeRaj-2401** [comment]
@vfsfitvnm this is some variant suggested by me to get rid of left navigation rail

![image](https://user-images.githubusercontent.com/88180887/197390736-834f08a8-70fd-4de6-a7c1-ef7e3e697371.png)

and u can use current player thumbnail for few swipe gestures such as **swipe left/right to go to previous/next song**

**Comment by scottiefalkon** [comment]
I like the full screen better.
Blur the full screen album art and place squared (shape optional) album
cover over top of it. Use swipe gestures as suggested.

I see no reason to make another music app that looks like every other app.

**Comment by sidsharths1245** [comment]
> @vfsfitvnm this is some variant suggested by me to get rid of left navigation rail (ignore icons inside and my bad drawing)
> 
> ![image](https://user-images.githubusercontent.com/88180887/197391186-897133b7-a1cd-431c-98f9-5b26eed588c7.png)
> 
> and u can use current player thumbnail for few swipe gestures such as **swipe left/right to go to previous/next song**

I actually love this idea, this seems quite practical. This way you can merge the functionality (gestures) of the new design with the old one by ditching the side bar.

**Comment by NeeRaj-2401** [comment]
> Another "more" practical version of this can be replacing the search icon with a burger menu icon and moving the search button inside, this way everything in the app can be controlled through this one single floating menu button.

this will lead too many clicks just to search song... and making user experience bad

**Comment by sidsharths1245** [comment]
> > Another "more" practical version of this can be replacing the search icon with a burger menu icon and moving the search button inside, this way everything in the app can be controlled through this one single floating menu button.
> 
> this will lead too many clicks just to search song... and making user experience bad

I think the trade offs are way better for just two extra click. The buttons will always be available whenever you need them without scrolling all the way to top or going back. It eliminates the need of a navigation bar/pane. It is small and doesn't clutter the UI. All the buttons/options are just one click away and you don't have to use two hands or stretch your thumb too much even if you have a big phone since the button is gonna be floating in the bottom right corner, it will be easily accessible for left handed folks as well. All in all this seems like a way better implementation.

**Comment by SecularSteve** [comment]
In my opinion there's only 1 change, and 1 addition I'd make to the current UI

**1. Having the buttons within a finger's reach.**

Basically, move the settings button, and the rest of them to the bottom of the screen for convenience on bigger phone screens. This way, I don't have to reach up with my second finger if I don't have to (maybe even the search bar text to the bottom, but that one should be discussed):

![Screenshot_20221023-182945~3.png](https://user-images.githubusercontent.com/33793273/197404205-282f5ece-cb58-450b-8506-c14b5084f6a7.png)

**2. Option to mirror the positions of the buttons. A "Mirror Mode" toggle.**

To make it more convenient if you're either left or right handed.

But that's just my opinion, and my suggestions.

**Comment by AloneWOLFx4** [comment]
1. You can make it so that the songs queue comes by swiping up instead of swiping left as it is more in line with all of the other music players people are used to.
2. This idea of album art full screen is a really good idea but its better for just the player screen and not the whole app background and making kind of an album art blur the background of the normal home, library stuff with a nav bar at the bottom.

This gestures idea is really cool tho having shortcuts instead of pressing the particular buttons.

Will the full screen lyrics be there all the time or toggled by the user ?

**Comment by AloneWOLFx4** [comment]
I don't know if you have used or not ... If you haven't you can use Tidal. It's got kind of a minimal UI if you want kind of like an inspiration for some parts of the UI

**Comment by vfsfitvnm** [comment]
> You can make it so that the songs queue comes by swiping up instead of swiping left as it is more in line with all of the other music players people are used to.

This is a perfectly fine alternative, but that would mean you need to swipe down to access the library. Not sure what's better

**Comment by Jak2k** [comment]
> > You can make it so that the songs queue comes by swiping up instead of swiping left as it is more in line with all of the other music players people are used to.
> 
> This is a perfectly fine alternative, but that would mean you need to swipe down to access the library. Not sure what's better

Tap for play/pause

Swipe left/right for next previous song
Hold and swipe left/right seeking in currenct song

Up for the queue (Currenctly it's the same so nothing changes
Down for the library (Currently also the same)

Long-Tap for Lyrics
Double-Tap for Like
Tripple-Tap for Nerd-Stats

**Comment by NeeRaj-2401** [comment]
Also add a gestures info in settings would be good idea :) atleast for new users
Cause some of them will unnecessarily open new issues like #489 and #495

**Comment by NeeRaj-2401** [comment]
> > Also add a gestures info in settings would be good idea :) atleast for new users
> 
> There could be an intro screen on the first launch that takes you through the gestures. You can make it skippable.

That will needs lots of efforts and time of developer...
Instead a normals gestures info somewhere within settings would be easier cause it will be similar to info/about section

**Comment by vfsfitvnm** [comment]
> That will unnecessarily need lots of efforts and time of developer... 

True, but it's necessary (so I can also avoid issues like #481). So there will be a welcome screen.

 I can also let the user decide whether to have the vertical setup (the original one I proposed) or the horizontal one (proposed by @AloneWOLFx4)

**Comment by Kreavita** [comment]
Hey, first of all thank you for making this thread and not just implementing a controversial design change without talking to the users, not everybody does that.

I like the app for its clean and modern look and its familiar feeling, therefore i prefer the player controls better the way they are implemented right now (there is a reason most audio players practically look the same ^^), but users could have it both ways if you added a ui style selection in the settings.

Some people mentioned the empty look of the player, so I would suggesst adding a very low opacity and blurred full screen cover to the background, but keep the regular cover the way it is, enabling lyrics could replace the cover and use the whole space.
Also i would prefer the triple dots button in the top right corner and imo the player queue button would look better if you replaced it with a gradient and changed its icon to small arrow up, but those are just minor complaints.

In terms of gestures, i like the ideas of @Jak2k, but i would like to still have all functionality available as buttons - lyrics and nerd stats could be hidden in the dots menu.

Regarding the rails / navbar problem: imo the solutions by @NeeRaj-2401 and @Sameky are more useable than the rails, also you could implement horizontal swiping to switch from albums to artists etc. - the rails look sleek but they feel weird to use for me.

Also i would love to have those little popup hints for each button when you long press it, this is implemented in most apps because sometimes the user cannot guess the function of each button just by its icon.

And thank you again for making this poll, and of course also for making this app - it is my go-to music player since i discovered it.

**Comment by Kreavita** [comment]
perhaps it would also improve the rails if all icons are shown the whole time, and the active rail would be in the accent color and maybe bold

**Comment by Iam-Eshwar** [comment]
I'm just suggesting u sir please once see the pulse music from playstore home screen design if possible make it as that app home screen design

**Comment by emkay5771** [comment]
Looks really interesting... How would it affect the rest of the app UI? As well, where would the shuffle button go? I noticed that in the current implementation, shuffle is hidden under 2 layers of interaction. Thanks for all of your work in this awesome project!

**Comment by AloneWOLFx4** [comment]
__![Screenshot_2022-10-25-09-53-58-863_com.aspiro.tidal.jpg](https://user-images.githubusercontent.com/88570159/197682221-201fc110-d625-4ede-a414-a1b8d26ecd3d.jpg)__

What can be done here is to swipe down on this music players screen to show up the library.
Swiping up may show the queue.

![Screenshot_2022-10-25-09-55-34-518_com.aspiro.tidal.jpg](https://user-images.githubusercontent.com/88570159/197682363-b8ccb538-f0db-4210-98eb-1088e496f52d.jpg)

This is an idea for lyrics page.
So instead of these being full screen they can be like a separate box under the player controls similiar to spotify or like it's in the current app as an overlay on the album art so it would appeal to people who like to see the visual elements of the song art and works.

**Comment by vfsfitvnm** [comment]
Guys, please put screenshots or large images urls inside this html block to make the post easier to read:
```<img src="URL_HERE" width="20%"/>```

**Comment by NeeRaj-2401** [comment]
Also @vfsfitvnm if u r thinking to implement swipe left/right on thumbnails for next/previous songs then these skip buttons will be almost of no use
<img src= "https://user-images.githubusercontent.com/88180887/197968771-e0d4c41d-ed90-4779-a9e8-1c1a3f0cc117.jpg" width="20%"/>
Then i think u should replace these skip buttons with 10s forward/backward to current playing song #70

**Comment by Khenxy** [comment]
The intended new UI concept looks really clean and visually pleasing I think it'll work well for the app. I've always thought the lyrics screen and text were tiny and now with more room I think the lyrics experience will get better. If you could also implement scrolling through the synchronised lyrics to seek the player to that timestamp on the song it would be awesome. Apart from that I think the whole gesture operated UI would be great. 
There should also be a three dot icon somewhere on the screen that you can click to expand a menu where you'd find more options like add to Playlist, more from artist, go to album etc...
Also if this is implemented a welcome screen would be necessary and maybe a pop up would appear the first few time you open the player screen giving hints on gestures and what they do (There should be a don't show again checkbox with this).
Also thanks for all the effort you've put into making this app, maintain it and listening to user feedback.

**Comment by vfsfitvnm** [comment]
Good points. They are not strictly correlated, however such redesign involves a change of the rest of the app as well - other doors may open. Also, bringing something new to the table fuels my motivation!

**Comment by Jak2k** [comment]
Maybe show one row with the most listened songs, one with the most listened albums/playlists and one with the most listened artists in "Home".

Each row could also have a see more button and options to choose other sorting methods (A-Z, Time since added, ...) or filtering methods (Only liked songs, Only Offline-available songs, ...)

**Comment by Iam-Eshwar** [comment]
Add download feature bro 🙏🙏🙏

**Comment by arfanmrafeek** [comment]
> Add download feature bro

The Dev is not interested in Download option, Because this is a Music Player not a mp3 downloader, Sorry for that. But you don't really need this because once you listen to a song, it's available in your cache, You can listen to it anytime ..

Whatever if you need mp3 for your PC etc.. 
Use downloader or there are other apps which has download option like Innertune, Blackhole, And Musify.

**Comment by ThatTakashi** [comment]
I prefer the current design of the app but I love the new concept of the full screen player with tiktok style song navigation

**Comment by Fatcakes** [comment]
@ThatTakashi I agree that the current design is better. The only thing the concept might look better on is android TV.

**Comment by vfsfitvnm** [comment]
Kind of yes, actually. Making the player the _first citizen_ and displaying the other screens on top of it allows those screen to **not** have their own "personality" (UI wise). That is, they can have a boring and generic design that doesn't require fancy components to stand out. This is just a theory, by the way :stuck_out_tongue:

**Comment by yc-codes** [comment]
Guys, take a look at **Poweramp**.
It has a great user base and i love it's UI

**Comment by selurvedu** [comment]
My 2¢:

The current UI is fine. It needs a few things improved here and there, but I like it as is.

**Comment by ishanveersg** [comment]
I don't like fact that I do not get to see the entire album art. I love album arts!

**Comment by xrdapp** [comment]
> As you all know, the version 0.5 brought a massive UI change. Some people are concerned about the ease of navigation, and I can't blame them (full discussion: #390). I honestly love how every screen looks like, but perhaps it's more good looking than functional. An aspect I don't like is the difficulty to tell _where_ you actually are: every screen resembles literally any other screen of the whole application.
> 
> The vertical navigation bar, also known as navigation rail, improved the UI, but harmed the usability (i.e. your finger has to move too much to change the content on the screen). I took some spare time to address the issues, but as soon as I removed the vertical bar (_the root of evil_) the screens looked arid - as they were just a pile of stuff glued together.
> 
> I'm not sure how these things matter in a music player app, however I always try to push things further so I started thinking about another design.
> 
> All music players have the same layout: bottom navigation bar (or navigation rail + horizontal tabs) + bottom player (that can be expanded). That's it. Thanks, material design.
> 
> I already do not use a top/bottom navigation component and now I'd like to see if I can get rid of the bottom player, too.
> 
> The concept I had in mind is having an _immersive_ player screen as the main screen of the application. With a lot of gestures.
> 
> Here it is:
> 
>  device-2022-10-22-165147.mp4 
> * Translucent theme - the playing song cover is the background of the whole app
> * Swipe up/down to change song
> * Swipe right to display the queue
> * Swipe left to access the library (not yet implemented)
> * Full screen lyrics (not yet implemented)
> 
> I'd like to know to hear what you think before investing a lot of time in this concept. It was fun to do 😋

For this I just want to give you some advice. This application will look good when the system and the basics in the music player are there and are also easier to find. You can also update about Playlists and profiles to make them more interesting to use.

**Comment by Iam-Eshwar** [comment]
It's ViMusic official telegram discussion group..
t.me/ViMusic_Official

**Comment by Iam-Eshwar** [comment]
![Screenshot_2022-10-30-23-55-24-711_org.thunderdog.challegram.jpg](https://user-images.githubusercontent.com/45062300/198895102-4d2a0604-e187-461e-a6f6-6059f6a13971.jpg)

**Comment by vfsfitvnm** [comment]
Little update: I'm actively pursuing this new concept. However, the UI evolved and it's slightly different from the one shown in the first video.

Also, having an immersive music player made me think about music players in general. 
People expect music players to behave like file managers: you can create, sort and organize files and directories. This is a completely logical thing to do in an era where songs are actually files you had to download in advance.

As I said [here](https://github.com/vfsfitvnm/ViMusic/pull/107):
> I believe music is temporary. That is I stick with an artist, a genre or a song for a period of time (a couple of days, one week, three months, one year) and then I just move on. It doesn't mean I only listen to few songs, but those few songs are the "cornerstones".
You know, music accompany our life, so it reflects our moods and what happen to us. That's why it is temporary, life changes and so the music. Yes, I will still listen to songs I previously enjoyed from time to time in the future, but they won't be my primary focus - my taste or life moved on.
I don't have a "all time favorites" playlist which I listen to regularly, because I already know it won't be listenable because it is too diversified.

A playlist attempts to fix your current mood forever. I never found them helpful as they have always become a "song graveyard". I honestly don't know why I implemented a playlist feature in the first place. 
"_Well, literally every music player have playlists, how can ViMusic not have them?_" I thought.

However, today we have a radio feature and recommendations. I think it's possible to build a music player just on top of that (and nothing else).
I think the proposed concept can really be the minimal, catalogue-less and immersive music player I pictured in my mind: no bookmarks, no albums[^1], no artists and, most importantly, no playlists. Yes trending songs, yes related songs and yes all-time favorites.

However, I will probably create a separate app (new name, new icon and so on) for this concept. Most importantly, these are just words. I still have to finish it and actually use it daily. Maybe I will realize it's dogshit...

[^1]: Maybe

**Comment by emkay5771** [comment]
> Little update: I'm actively pursuing this new concept. However, the UI evolved and it's slightly different from the one shown in the first video.
> 
> Also, having an immersive music player made me think about music players in general. People expect music players to behave like file managers: you can create, sort and organize files and directories. This is a completely logical thing to do in an era where songs are actually files you had to download in advance.
> 
> As I said [here](https://github.com/vfsfitvnm/ViMusic/pull/107):
> 
> > I believe music is temporary. That is I stick with an artist, a genre or a song for a period of time (a couple of days, one week, three months, one year) and then I just move on. It doesn't mean I only listen to few songs, but those few songs are the "cornerstones".
> > You know, music accompany our life, so it reflects our moods and what happen to us. That's why it is temporary, life changes and so the music. Yes, I will still listen to songs I previously enjoyed from time to time in the future, but they won't be my primary focus - my taste or life moved on.
> > I don't have a "all time favorites" playlist which I listen to regularly, because I already know it won't be listenable because it is too diversified.
> 
> A playlist attempts to fix your current mood forever. I never found them helpful as they have always become a "song graveyard". I honestly don't know why I implemented a playlist feature in the first place. "_Well, literally every music player have playlists, how can ViMusic not have them?_" I thought.
> 
> However, today we have a radio feature and recommendations. I think it's possible to build a music player just on top of that (and nothing else). I think the proposed concept can really be the minimal, catalogue-less and immersive music player I pictured in my mind: no bookmarks, no albums[1](#user-content-fn-1-e7c0e214c776915ecd14c6c52259a9f8), no artists and, most importantly, no playlists. Yes trending songs, yes related songs and yes all-time favorites.
> 
> However, I will probably create a separate app (new name, new icon and so on) for this concept. Most importantly, these are just words. I still have to finish it and actually use it daily. Maybe I will realize it's dogshit...
> 
> ## Footnotes
> 1. Maybe [↩](#user-content-fnref-1-e7c0e214c776915ecd14c6c52259a9f8)

Interesting idea... Conceptually, would it not be similar to Spotify Stations, an app that they retired in the past year or so? It could be a fascinating experience if done right, with some customization available.

**Comment by NeeRaj-2401** [comment]
> today we have a radio feature and recommendations. I think it's possible to build a music player just on top of that (and nothing else).
I think the proposed concept can really be the minimal, catalogue-less and immersive music player I pictured in my mind: no bookmarks, no albums, no artists and, most importantly, no playlists. Yes trending songs, yes related songs and yes all-time favorites.

Agreeing with ur thoughts on people treating music apps as file manager...

The Proposed concept Seems very interesting 🙌, 
i.e. a music app just to listen songs. No need to store and manage playlists, artist, songs lists etc . Just open the app listen to whatever we want (minimal)..

 This can lead to a very unique music streaming app so far.....

**Comment by ikanakova** [comment]
> no bookmarks, no albums1, no artists and, most importantly, no playlists. Yes trending songs, yes related songs and yes all-time favorites

I think the option without playlists and bookmarks etc. can be a great thing for those who listen to currently popular music, but I can't imagine that it would also suit those with specific tastes.

> You can keep the general concept you're proposing in place, but please include the ability to play entire albums, and ideally allow for a way to bookmark them, so that they can easily be revisited and listened to.

I also think that it would be ideal if bookmarks for albums and possibly artists keep in the application, then I think everyone would be satisfied.

**Comment by Jak2k** [comment]
# Playlists
Please keep the playlists. I liked all songs I like, but added all song I love to a playlist. So I have "two sorts of likes".

It would be great to keep the feature of playlists. Maybe make it optional...

# Artists
Please keep artists. Of course, the music you hear changes, but there are always artists, you can always hear.

# Recommendations
Yes, they are really good today, but they are intransparent. Maybe you could display, why the song is recommendet.

## Idea for recommendations

1. Give every *artist, album, song and playlist* a **score** base on *playtime, click-count, like-count, skip-percentage, bookmarked or not*
2. The **final score** is calculated by taking the score of the artist(s), playlist(s), similar songs and album. Score get lesser over time.
3. *Higher scored songs* are more often recommended.
4. If the **score is to less, it won't be recommended**.

**Comment by ARBoyGo** [comment]
@vfsfitvnm 

> However, I will probably create a separate app (new name, new icon and so on) for this concept. Most importantly, these are just words. I still have to finish it and actually use it daily. Maybe I will realize it's dogshit...

I'm not against it, Minimalistic music player, But people need so many features, In case you create one, Maybe it ViMusic lite, Better no hard time in logo making and etc..

But this make me feel you will drop this project in mid way, In case please notify us, Maybe other developer use it for future of ViMusic.

**Comment by ARBoyGo** [comment]
> A playlist attempts to fix your current mood forever. I never found them helpful as they have always become a "song graveyard". I honestly don't know why I implemented a playlist feature in the first place.
"Well, literally every music player have playlists, how can ViMusic not have them?" I thought.

I have many playlist, The playlist I'll open will be according to my current mood, Feeling happy I play those, Depressed I play those, I have playlist for all kind of taste from West to east, More than 5-8 languages to organise I need playlist, In my case I use it very much, and a plus point, I even have playlist for my bed time, Just play go to sleep, Now we have loop none option, I keep my phone in charge and play music in Bluetooth devices and head to bed. It's just great.

**Comment by NeeRaj-2401** [comment]
@vfsfitvnm busy these days? Or working on new concepts.. cause .. most of us including me wants to see the progress of new concept .. can u please show some recording etc or alph apk..?

**Comment by 825i** [comment]
Commenting to the OP. To be honest, I hate it.

First of all (and please don't take this the wrong way). I already don't understand your choices with the current layout. It's utterly confusing. Some small tweaks to it could drastically improve the quality of life, with no need for such a total overhaul.

Let's face it. What you are suggesting is that you get to work on completely ripping the app UI apart. Some might like the change, some likely will not. So you'll be stuck feeling like a good chunk of your time has been wasted on a highly time consuming feat.

So why not just improve what you have already worked hard on? Some minor tweaks like eg. putting a flashing cursor-bar where the search area is at the top of the screen on the current design. That would be great, it would actually let you know that you can type there. You could even darken that text area a little to provide even higher readability.

The thing is, a lot of the choices you already made are really not good for partially blind/color-blind/disabled people. I am none of those but a lot of people are. So it doesn't make any sense to make a full new UI that is even more minimalist.

Fix mistakes rather than throwing the baby out with the bathwater.

That's my 2cents.

P.S. I love the recommendations in this app. They are literally as good as Apple Music's.

**Comment by 825i** [comment]
I realise I could edit my above post but it's already long enough.

I also see a lot of off-topic posts on in this issue. Feature requests, questions about how to use the app etc. which could be figured out by simply trying things out. 

The devs have enough noise to cut through as it is. So let's keep it on topic. Provide feedback about the UI changes and leave it at that. If you want to have a big discussion about things, probably best to make another ticket/thread,

**Comment by legalizenukes** [comment]
I'd like start off my comment by saying that I really like ViMusic. Compared to the countless other music streaming apps I've used, it is by far the most lightweight and doesn't sacrifice functionality or niche features like Wavelet support (which I cannot do without) while being so. The caching feature is also one that I find more useful every day. 

However, I do agree that the user interface can be a bit difficult to navigate both in terms of one-handed-usability and at times can be "not-fluid". 

Since it is difficult convey UI ideas through words alone, I've made a mockup of ViMusic's "Now Playing" screen and tried to address the issues that I've faced while using that app (and I'm sure others have faced similar issues as well).

============================================================================================

<img src="https://user-images.githubusercontent.com/118522035/202672382-b49563ff-6cc9-467a-818e-4bac9d2818a9.jpg" width=25% height=25%>   <img src="https://user-images.githubusercontent.com/118522035/202674890-95815039-1f89-485d-a472-d7fd95b9df1a.jpg" width=25% height=25%>   <img src="https://user-images.githubusercontent.com/118522035/202674976-3a99ae46-e546-4371-9b20-c963f3f76989.jpg" width=25% height=25%> 

Above are three screenshots taken from my phone of the original ViMusic UI (OnePlus 7 Pro, 2340 x 1080 resolution, 320 dp, immersive mode enabled).

<img src="https://user-images.githubusercontent.com/118522035/202690956-a15e72c8-511a-4e20-9c18-2783325ff037.jpg" width=25% height=25%>   <img src="https://user-images.githubusercontent.com/118522035/202690965-da2a3e6c-ef1c-4a0f-92a0-e5a361b621e5.jpg" width=25% height=25%>   <img src="https://user-images.githubusercontent.com/118522035/202690972-effefae2-9b05-4c7c-b70b-ae8095dce304.jpg" width=25% height=25%> 

Above are my mockup redesigns (3120 x 1440 resolution). The padding on some elements could be tweaked along with some other things, but this is what I came up with so far.

============================================================================================

Now I would like to quickly go over how the user interactions would carry out from the top down, left to right. Hopefully it is coherent.

First and foremost, I'm a big fan of haptic feedback (makes the software more lifelike), one-handed usability, and simple gestures so much of my design choices are based on these preferences. The smaller icons being "Like, Lyrics, "Addtional Actions", and "Repeat" will have haptic feedback weaker than the level used for the "Progress Bar", "Back/Forward Buttons", "Play/Pause Button", "Up Next/Now Playing Bar", and "Bottom Bar". All icons will also generate a small circle behind them on interaction similar to the way they do now. Haptic feedback should be toggleable in the settings.

Cover Art: A single tap on the cover art will bring the user to the album or single the song belongs to. A long press and hold will show the metadata similarly to how it already works now. A swipe left or right from the cover art will go back or forward a song respectively. The cover art will slightly follow the direction of the finger as this gesture takes place. Cover art generated for YouTube videos are not zoomed in, just cropped into a square.

Song Title & Artist Name: A single tap on the song title or artist name will bring you to the artist's page. A long tap on these elements will bring up a menu similar to the one in the second mockup except it will be centered. The names of ALL artists (main artist + features) will be listed and their circular profile pictures will be shown and the text/image elements left aligned. (No mockup yet for this unfortunately).

Like: User presses the icon, it is filled with a red color, slightly expands, and a vibration is made. User releases the icon, it remains red, shrinks back to normal size, another weaker vibration is made. Unliking a song performs the same actions except the color fill is removed

Lyrics: Works similarly to like icon except it fills with white instead of red. It will stay filled as long as the lyrics are shown. Tapping the icon will hide some of the song information and reveal the lyrics. The animation for this could just be a circle that expands in size from the center of the icon to reveal the box containing the lyrics. During this animation the Like and Lyrics icons are also moved slightly down. Closing the lyrics performs the reverse animation.

Lyrics Box: If the song supports synced lyrics then previous or future lyrics will be gray and the current verse will be white. The user will still be able to scroll while synced lyrics are on, allowing them to seek ahead or reread missed lyrics. However, when the current lyrics are close enough to the center of the screen it will snap back to being fully synced and moving on its own.

Progress Bar: Works similar to the way it works now functionally. Seeking using the bar will slightly thicken it as it does now, remove the circle, and also give some haptic feedback. When this happens the elasped time/total time will temporarily fade away and a small bubble similar in "material" to the additional actions menu will appear over the user's finger and show them the exact time they are seeking to. When the new time is selected the progress bar will return to its normal look and give a slightly weaker vibration than the initial one. (No mockup for the bubble yet).

Additional Actions: Similar haptic feedback system to the other smaller buttons. Tapping this will bring up a menu with additional options. The options themselves work the same way as before. Reveal animation is also a circle like the lyrics box. Tapping anywhere else after this menu or pressing back will hide it.

Back/Forward: Work the same way they do now.

Play/Pause: Works the same way it does now. 

Up Next/Now Playing: Previews the song that is coming up. Tapping this will bring up the queue menu. Moving to a DIFFERENT menu than the menu with the cover art will replace the upcoming song information with information on the current song instead. The box gains a progress bar on the top along with smaller play/pause and back/forward buttons similar to how it is now. The "Up next" text and box will be removed. Tapping the "Up Next/Now Playing" box in this scenario will bring the user to menu with the cover art. Has stronger type haptics like the Play/Pause button.

Bottom Bar: In order has the following menus: "Picks" (recommended), "Music" (library), "Search" (search), "Discover" (includes the catalogueless, immersive music player you had in mind), "Settings" (settings). Also has strong type haptic feedback. An icon brightens when the user is in it's respective menu and darkened when moving to a different menu. The text underneath slides and fades/animates into the new menu title when a new menu is tapped. 

Picks: Picks will simply show recommended artists, songs, albums, and playlists based on what you've listened to before. 

Music: Has saved/liked artists, songs, albums, and playlists. Also a section for songs cached for offline playback and recent songs with similar categories. I haven't fleshed out the concept for this just yet.

Search: Pretty self-explanatory, works similarly to how it does now except the results are slightly rearranged to match the layout of the "Music" menu. 

Discover: Includes the catalogueless, immersive music player you had in mind. I'm not sure if that means it shows completely random music for one to discover, but whatever the concept becomes I think it's a cool concept that should be a part of ViMusic. Would be great when the songs I listen to get stale and to find new genres.

Settings: Brings up the settings menu. Just rearranged into one page with categories. 

============================================================================================

Because this is a mockup and I don't have any coding limitations I'm not sure how much of this UI is feasible. I am not a developer. But hopefully it's a cool idea that addresses some of the current issues with ViMusic. Feel free to use all of this, some of this, or none of it, totally up to you. At the end of the day your vision of an ideal music player UI may be much different, I'll continue to use ViMusic regardless. If you would like me to draft some mockups for some the stuff I didn't get to, in time I would be happy to make those as well. Would like to hear your thoughts.

**Comment by shpatter** [comment]
I haven't read anything in this discussion, but please leave the option to keep the current ui, it's very good.

**Comment by vfsfitvnm** [comment]
@legalizenukes I apologize for the late reply, but thanks for the effort you put in it - that made me smile - that's the perfect example of how an appreciated external contribution should look like.

However, I feel like the "minimal, catalog-less and immersive" music player I am working on will take place on another repository (with a new icon, new name...), as I'm literally writing another app from scratch. That said, once the new app is finished, I will start using it daily and I will uninstall ViMusic, eventually.

This repository is already on maintenance mode (i.e. only bug fixes are being processed / worked on). Unless a major and unexpected event happens, I don't think I will add features or redesign it.

**Comment by scottiefalkon** [comment]
Looking forward to the new player. It sounds very interesting.

**Comment by scottiefalkon** [comment]
I second that. The fact that you're doing something new and different is
admirable. I love the concept and have no doubt that it will turn out
great.

**Comment by vfsfitvnm** [comment]
For the curious ones, I can only share two screenshots of the new single-screen music player:

<p float="left">
<img src="https://user-images.githubusercontent.com/46219656/203121822-9e364b97-f5db-407b-b85d-0f7f855cbb84.jpg" width="25%"/>
<img src="https://user-images.githubusercontent.com/46219656/203121837-9b5d1ee1-0e81-4bf0-97b1-72892dae0815.jpg" width="25%"/>
</p>

Of course this is still unfinished (I didn't test "on person" yet). The development pace is quite slow (this is a good sign), so I can't share an ETA I'm afraid.

Regarding the features I already have clear in mind:
- Built with minimalism and gesture-ism in mind.
- Vertical scroll to change song, tap to play/pause.
- There's only a single song list: the "favorites" one. The heart button acts as a "add to library".
- When the app is opened, you can immediately play the trending songs (the ones you have been listening to for most of the recent time) as a queue, or start the radio from one of them.
- You can view the lyrics, album or stats of the playing song.
- You can only search for songs or videos. However, I _may_ support artists and albums, but they won't have a dedicated page. That said, the album itself or the artist radio will start playing when tapped.
- No playlists at all.
- No bookmark feature.

**Comment by scottiefalkon** [comment]
See. I think it looks great so far and love the simplicity of it. There are
so many other players that do what Vi does. And I love Vi. But, I very
often start radio stations off of a song and go with the flow after that.
I'm not a big playlist type person and though I listen to albums I prefer
just starting a station based off of a song. I could see adding Artists.
But, I don't think albums is necessary.

I'm really looking forward to this. A visualizer across the center part, in
a banner type form, with a translucent background like the other items
shown would be a cool option to be able to enable or disable.

**Comment by slimfish08** [comment]
@legalizenukes I really love this suggestion, very useful and looks great. It could be added following the current UI Design, like no transparency and keeping the original player's button, but so far it is great. 

I came to GitHub specifically to request a folder structure for Playlists in this app, as I use them a lot mostly to save and listen to music from different countries in different languages and songs I love from a similar genre. But I decided to read this thread beforehand. @vfsfitvnm thanks for your great work in this app. I left Spotify and YT Music for this amazing app.

I'm looking forward for the other app ('DiMusic' as in Discover maybe?) to find new songs easily. It's a great concept, and different.

**Comment by NeeRaj-2401** [comment]
@vfsfitvnm your concept and design in shared screenshot is amazing.. can't wait to use that new app 🙌
for those who wants album, library, artist etc all of the features they can go with ViMusic
and for the people who likes your new concept of a music app will love to use that.

**Comment by Bazixv13** [comment]
In my opinion You give an setting to change between ui the old one and new

**Comment by scottiefalkon** [comment]
This has already been settled by the Dev. New app with a new UI and
different music experience. Let my man do his thing. He has said that the
current app is in maintenance mode only. Bug fixes only. He is
concentrating on the new app. Be thankful for what he is giving you with
this app. And show him support for his upcoming project.

**Comment by tvnmguy** [comment]
![img_2022_11_25_13_55_48.jpg](https://user-images.githubusercontent.com/116799295/203935641-c517eb8b-fa24-454c-9ef0-886e48507f95.jpg)


Just for a fun😉

**Comment by Anonymous1231-dev** [comment]
> For the curious ones, I can only share two screenshots of the new single-screen music player:
> 
> <img alt="" width="25%" src="https://user-images.githubusercontent.com/46219656/203121822-9e364b97-f5db-407b-b85d-0f7f855cbb84.jpg"> <img alt="" width="25%" src="https://user-images.githubusercontent.com/46219656/203121837-9b5d1ee1-0e81-4bf0-97b1-72892dae0815.jpg">
> 
> Of course this is still unfinished (I didn't test "on person" yet). The development pace is quite slow (this is a good sign), so I can't share an ETA I'm afraid.
> 
> Regarding the features I already have clear in mind:
> 
> * Built with minimalism and gesture-ism in mind.
> * Vertical scroll to change song, tap to play/pause.
> * There's only a single song list: the "favorites" one. The heart button acts as a "add to library".
> * When the app is opened, you can immediately play the trending songs (the ones you have been listening to for most of the recent time) as a queue, or start the radio from one of them.
> * You can view the lyrics, album or stats of the playing song.
> * You can only search for songs or videos. However, I _may_ support artists and albums, but they won't have a dedicated page. That said, the album itself or the artist radio will start playing when tapped.
> * No playlists at all.
> * No bookmark feature.

@vfsfitvnm Wow, it's gorgeous! But will the lyrics feature still implemented in your new app?

**Comment by Anonymous1231-dev** [comment]
> 

Wow, it's gorgeous! @vfsfitvnm i think the new app name can be "ImmerseTune"?

**Comment by Anonymous1231-dev** [comment]
> For the curious ones, I can only share two screenshots of the new single-screen music player:
> 
> <img alt="" width="25%" src="https://user-images.githubusercontent.com/46219656/203121822-9e364b97-f5db-407b-b85d-0f7f855cbb84.jpg"> <img alt="" width="25%" src="https://user-images.githubusercontent.com/46219656/203121837-9b5d1ee1-0e81-4bf0-97b1-72892dae0815.jpg">
> 
> Of course this is still unfinished (I didn't test "on person" yet). The development pace is quite slow (this is a good sign), so I can't share an ETA I'm afraid.
> 
> Regarding the features I already have clear in mind:
> 
> * Built with minimalism and gesture-ism in mind.
> * Vertical scroll to change song, tap to play/pause.
> * There's only a single song list: the "favorites" one. The heart button acts as a "add to library".
> * When the app is opened, you can immediately play the trending songs (the ones you have been listening to for most of the recent time) as a queue, or start the radio from one of them.
> * You can view the lyrics, album or stats of the playing song.
> * You can only search for songs or videos. However, I _may_ support artists and albums, but they won't have a dedicated page. That said, the album itself or the artist radio will start playing when tapped.
> * No playlists at all.
> * No bookmark feature.

Wow, it's gorgeous! @vfsfitvnm i think the new app name can be "ImmerseTune"?

**Comment by NeeRaj-2401** [comment]
> "ImmerseTune"

sounds too similar, Inspired by innerTune

i think name new name suggestions are welcome but vfsfitvnm will himself come with some unique one as he came up and stick with ViMusic

**Comment by vfsfitvnm** [comment]
I had zero ideas when it came to deciding on the app name, so I made the icon first. I noticed it vaguely resembled a "V", hence ViMusic (lol)

**Comment by NeeRaj-2401** [comment]
> I had zero ideas when it came to deciding on the app name, so I made the icon first. I noticed it vaguely resembled a "V", hence ViMusic (lol)

In that case **ViMusicLite** would be better choice because
1. as it goes with the **minimalistic** features of ur app 
2. the idea of that project came up during the original Vimusic development

**Comment by vfsfitvnm** [comment]
No, it will certainly have a different name

**Comment by NeeRaj-2401** [comment]
as some of us are very excited for the new concept and design, Can u please show us the progress ?

**Comment by rex50** [comment]
> No, it will certainly have a different name

Is there anyway to support you with developement side on the new project? Or are you going lone wolf?

**Comment by vfsfitvnm** [comment]
> Is there anyway to support you with developement side on the new project? Or are you going lone wolf?

Kind of yes:

- The majority of the UI is already implemented. Other than refinishing many little details, polishing the code and rewriting things, I still have to figure out _where_ and _how_ to show the current playing track lyrics, stats for nerds and album. That is deciding gestures, animations and whatsoever.
- I'd like to get rid of ExoPlayer. You know, for better customization and learning purposes.
- I'd like to host the playback service into a separate process - so we don't have to worry about requesting battery optimizations. Unfortunately it requires a lot of work (especially due to IPC), so I leave this one for last.

**Comment by NeeRaj-2401** [comment]
> * The majority of the UI is already implemented

Can u provide some glimpse of it... So that we can think and come up with some gesture ideas

**Comment by NeeRaj-2401** [comment]
> <img alt="" width="25%" src="https://user-images.githubusercontent.com/46219656/203121822-9e364b97-f5db-407b-b85d-0f7f855cbb84.jpg"> <img alt="" width="25%" src="https://user-images.githubusercontent.com/46219656/203121837-9b5d1ee1-0e81-4bf0-97b1-72892dae0815.jpg">

considering this UI
1. Lyrics : one tap on thumbnail (lyrics shown on somewhere between middle of whole thumbnail)
     also if u r thinking to implement synchronized lyrics then in lyrics section one single button would help
    
2. stats of nerds : tap and hold  (stats of nerds, i personally dont use but still if U wish to)
3.  double tap on  thumbnail to pause/play song
4. miniplayer tap or swipe up  to expand it (which shows progress bar and upcoming songs by horizontal swipe)
 also i noticed a arrow button in expanded mini player i guess that would help to see more upcoming songs at once
5. swipe left and right are still vacant so albums and other stuffs as u wish to

most of above are already in current ViMusic but according to me these gestures would be user friendly and good to go with


Still Other ideas or additions would be great !

**Comment by vgovind94** [comment]
As much as I like the user interface, I believe that it sways a bit away from the Google's material design. The current UI is well and good. Vertical navigation can have a learning curve but it sure is more aesthetically pleasing. Also, swiping down to change song can be irritating at times as it is mostly used to see the song queue, lyrics etc. 

Also, instead of changing the song for each swipe, you could add the lyrics functionality to it. Swipe up and the lyrics show up.  Left and right swipes can be used for moving between music. Double tap on the artist image could trigger this immersive display. This way users would have more choice.

**Comment by vgovind94** [comment]
> I'd like start off my comment by saying that I really like ViMusic. Compared to the countless other music streaming apps I've used, it is by far the most lightweight and doesn't sacrifice functionality or niche features like Wavelet support (which I cannot do without) while being so. The caching feature is also one that I find more useful every day.
> 
> However, I do agree that the user interface can be a bit difficult to navigate both in terms of one-handed-usability and at times can be "not-fluid".
> 
> Since it is difficult convey UI ideas through words alone, I've made a mockup of ViMusic's "Now Playing" screen and tried to address the issues that I've faced while using that app (and I'm sure others have faced similar issues as well).
> 
> ============================================================================================
> 
> <img alt="" width="25%" height="25%" src="https://user-images.githubusercontent.com/118522035/202672382-b49563ff-6cc9-467a-818e-4bac9d2818a9.jpg"> <img alt="" width="25%" height="25%" src="https://user-images.githubusercontent.com/118522035/202674890-95815039-1f89-485d-a472-d7fd95b9df1a.jpg"> <img alt="" width="25%" height="25%" src="https://user-images.githubusercontent.com/118522035/202674976-3a99ae46-e546-4371-9b20-c963f3f76989.jpg">
> 
> Above are three screenshots taken from my phone of the original ViMusic UI (OnePlus 7 Pro, 2340 x 1080 resolution, 320 dp, immersive mode enabled).
> 
> <img alt="" width="25%" height="25%" src="https://user-images.githubusercontent.com/118522035/202690956-a15e72c8-511a-4e20-9c18-2783325ff037.jpg"> <img alt="" width="25%" height="25%" src="https://user-images.githubusercontent.com/118522035/202690965-da2a3e6c-ef1c-4a0f-92a0-e5a361b621e5.jpg"> <img alt="" width="25%" height="25%" src="https://user-images.githubusercontent.com/118522035/202699939-e122584d-cfa2-4b60-b1ed-ebcf22002071.jpg">
> 
> Above are my mockup redesigns (3120 x 1440 resolution). The padding on some elements could be tweaked along with some other things, but this is what I came up with so far.
> 
> ============================================================================================
> 
> Now I would like to quickly go over how the user interactions would carry out from the top down, left to right. Hopefully it is coherent.
> 
> First and foremost, I'm a big fan of haptic feedback (makes the software more dynamic), one-handed usability, and simple gestures so much of my design choices are based on these preferences. The smaller icons being "Like, Lyrics, "Addtional Actions", and "Repeat" will have haptic feedback weaker than the level used for the "Progress Bar", "Back/Forward Buttons", "Play/Pause Button", "Up Next/Now Playing Bar", and "Bottom Bar". All icons will also generate a small circle behind them on interaction similar to the way they do now. Haptic feedback should be toggleable in the settings.
> 
> Cover Art: A single tap on the cover art will bring the user to the album or single the song belongs to. A long press and hold will show the metadata similarly to how it already works now. A swipe left or right from the cover art will go back or forward a song respectively. The cover art will slightly follow the direction of the finger as this gesture takes place. Cover art generated for YouTube videos are not zoomed in, just cropped into a square.
> 
> Song Title & Artist Name: A single tap on the song title or artist name will bring you to the artist's page. A long tap on these elements will bring up a menu similar to the one in the second mockup except it will be centered. The names of ALL artists (main artist + features) will be listed and their circular profile pictures will be shown and the text/image elements left aligned. (No mockup yet for this unfortunately).
> 
> Like: User presses the icon, it is filled with a red color, slightly expands, and a vibration is made. User releases the icon, it remains red, shrinks back to normal size, another weaker vibration is made. Unliking a song performs the same actions except the color fill is removed
> 
> Lyrics: Works similarly to like icon except it fills with white instead of red. It will stay filled as long as the lyrics are shown. Tapping the icon will hide some of the song information and reveal the lyrics. The animation for this could just be a circle that expands in size from the center of the icon to reveal the box containing the lyrics. During this animation the Like and Lyrics icons are also moved slightly down. Closing the lyrics performs the reverse animation.
> 
> Lyrics Box: If the song supports synced lyrics then previous or future lyrics will be gray and the current verse will be white. The user will still be able to scroll while synced lyrics are on, allowing them to seek ahead or reread missed lyrics. However, when the current lyrics are close enough to the center of the screen it will snap back to being fully synced and moving on its own.
> 
> Progress Bar: Works similar to the way it works now functionally. Seeking using the bar will slightly thicken it as it does now, remove the circle, and also give some haptic feedback. When this happens the elasped time/total time will temporarily fade away and a small bubble similar in "material" to the additional actions menu will appear over the user's finger and show them the exact time they are seeking to. When the new time is selected the progress bar will return to its normal look and give a slightly weaker vibration than the initial one. (No mockup for the bubble yet).
> 
> Additional Actions: Similar haptic feedback system to the other smaller buttons. Tapping this will bring up a menu with additional options. The options themselves work the same way as before. Reveal animation is also a circle like the lyrics box. Tapping anywhere else after this menu or pressing back will hide it.
> 
> Back/Forward: Work the same way they do now.
> 
> Play/Pause: Works the same way it does now.
> 
> Up Next/Now Playing: Previews the song that is coming up. Tapping this will bring up the queue menu. Moving to a DIFFERENT menu than the menu with the cover art will replace the upcoming song information with information on the current song instead. The box gains a progress bar on the top along with smaller play/pause and back/forward buttons similar to how it is now. The "Up next" text and box will be removed. Tapping the "Up Next/Now Playing" box in this scenario will bring the user to menu with the cover art. Has stronger type haptics like the Play/Pause button.
> 
> Bottom Bar: In order has the following menus: "Picks" (recommended), "Music" (library), "Search" (search), "Discover" (includes the catalogueless, immersive music player you had in mind), "Settings" (settings). Also has strong type haptic feedback. An icon brightens when the user is in it's respective menu and darkened when moving to a different menu. The text underneath slides and fades/animates into the new menu title when a new menu is tapped.
> 
> Picks: Picks will simply show recommended artists, songs, albums, and playlists based on what you've listened to before.
> 
> Music: Has saved/liked artists, songs, albums, and playlists. Also a section for songs cached for offline playback and recent songs with similar categories. I haven't fleshed out the concept for this just yet.
> 
> Search: Pretty self-explanatory, works similarly to how it does now except the results are slightly rearranged to match the layout of the "Music" menu.
> 
> Discover: Includes the catalogueless, immersive music player you had in mind. I'm not sure if that means it shows completely random music for one to discover, but whatever the concept becomes I think it's a cool concept that should be a part of ViMusic. Would be great when the songs I listen to get stale and to find new genres.
> 
> Settings: Brings up the settings menu. Just rearranged into one page with categories.
> 
> ============================================================================================
> 
> Because this is a mockup and I don't have any coding limitations I'm not sure how much of this UI is feasible. I am not a developer. But hopefully it's a cool idea that addresses some of the current issues with ViMusic. Feel free to use all of this, some of this, or none of it, totally up to you. At the end of the day your vision of an ideal music player UI may be much different, I'll continue to use ViMusic regardless. If you would like me to draft some mockups for some the stuff I didn't get to, in time I would be happy to make those as well. Would like to hear your thoughts.

Beautiful designs. Sort of like a fusion between Apple Music and Spotify. Also, could be great if you could add some Neumorphism elements to the UI.

**Comment by Rhoades1** [comment]
Hello everyone. I just recently discovered the app but I already use it a lot. I state that English is not my first language, so please forgive the mistakes. But now let's move on to us: I've read a lot of discussions and a lot of comments, which I think is essential for getting to know a new app and solving the first "problems" without creating useless discussions already covered.

From my point of view the app as it is now is very good. It's elegant, clean and tidy. Personally I don't have big problems with the vertical bar (about this, in my opinion it could be improved by making it slightly darker than the rest of the screen, including the writing and the icons above them. I also venture that the writing could be placed horizontally, maybe slightly smaller, but I don't push much on this because I should see them first to give a "definitive" opinion).

Many of us create Playlists and I think it would be very useful to have a "search" button within each of them, so as to find a song first rather than always scrolling. In my opinion it would also be a good thing, for example, once inside a playlist, to have the name of the playlist more highlighted.

I also add, as far as I'm concerned, that after a while this discussion became confusing and I don't understand if the new version will have similar ideas to the first video posted by vfsfitvnm (I'm referring to the screen with only the cover and then the use of gestures ) or whether it will be implemented directly in a new app.

I give my opinion: if there will be a makeover of the app's appearance like in the opening video, I think it's very nice and minimal, but every concept shouldn't be exaggerated. Personally, always referring to the video, I would move the player of the song slightly higher; many of us use gestures and it could create conflict. Furthermore, I would not like, as soon as I open the app, to immediately enter a song screen. Instead, the use of gestures is welcome: I would suggest the following:

- from bottom to top: next song
- from top to bottom: previous song
- from left to right: opening (not necessarily full screen, even a small part like the one occupied by the current vertical menu is enough) in which to see the items already present in the vertical menu, but perhaps written horizontally.
- from right to left: I would like to see the song queue
- touch screen: start/pause song
- keys: I would leave few; in addition to the song's minute bar, a favorites button, one to add the song directly to one of the playlists and then the classic three dots, keeping the current menu which seems complete to me.

VERY IMPORTANT: One thing that is also missing in the current version of the app: when I see the queue of songs I think it's more convenient for the song I'm listening to to be highlighted more, and I would like this to happen even if I go to the playlist where this song is featured, including the "offline" section.

I read that someone wants the cover of the slightly blurry song: to give an opinion I should see two photos, one of the blurry version and one of the normal version. Still watching the opening video, I notice other icons at the top of the screen (wifi, battery, etc.). These with a non-blurred cover look like part of the app, so in that case it would be useful to blur a bit, or in any case invent something, because one thing and having those icons in the current version of the app (where the background is a single color), it is one thing to have them on the covers, which will be the most disparate, making it a bit messy and confusing.

Ok, I think I've written everything, I hope whoever works on the app reads (it's a bit long I know, you'll forgive me) and above all I hope it's understandable: as I said, I'm not a native English speaker.

I await developments, thank you so much for the contributions and for the fantastic app, and greetings :)

**Comment by rex50** [comment]
@vfsfitvnm I was busy somewhere else.

>  I still have to figure out where and how to show the current playing track lyrics, stats for nerds and album. That is deciding gestures, animations and whatsoever.

Here are few ideas for button placements and UI. Refinement is required but yeah this what I have in mind.


| Lyrics style 1                                                                                                         | Lyrics style 2                                                                                                         | New Button UI                                                                                                         | 
|------------------------------------------------------------------------------------------------------------------------|------------------------------------------------------------------------------------------------------------------------|-----------------------------------------------------------------------------------------------------------------------|
| <video src='https://user-images.githubusercontent.com/12392404/205430695-505f68cd-0ec8-4525-ab6c-27436339807b.webm' /> | <video src='https://user-images.githubusercontent.com/12392404/205430700-65d6871c-ba82-4f9b-b7af-dff31d21261d.webm' /> | <video src='https://user-images.githubusercontent.com/12392404/205430702-55467701-b813-4624-ad41-882a10684a12.webm'/> |


>     * I'd like to get rid of ExoPlayer. You know, for better customization and learning purposes.
This makes sense as long as you wanna learn and explore options. I too wanna learn these stuff in detail for better understanding of the media plaback. But if you are in hurry in releasing a new version of the app then I won't recommend.


>     * I'd like to host the playback service into a separate process - so we don't have to worry about requesting battery optimizations. Unfortunately it requires a lot of work (especially due to IPC), so I leave this one for last.
Recently I have worked on audio playback in seperate service using Exoplayer (can share some insights). It works absolutely fine but I'm not happy the way I've implemented. So definetly need to explore.

**Comment by Anonymous1231-dev** [comment]
> @vfsfitvnm I was busy somewhere else.
> 
> ##
> 
> >  I still have to figure out where and how to show the current playing track lyrics, stats for nerds and album. That is deciding gestures, animations and whatsoever.
> 
> Here are few ideas for button placements and UI. Refinement is required but yeah this what I have in mind.
> 
> 
> | Lyrics style 1                                                                                                         | Lyrics style 2                                                                                                         | New Button UI                                                                                                         | 
> |------------------------------------------------------------------------------------------------------------------------|------------------------------------------------------------------------------------------------------------------------|-----------------------------------------------------------------------------------------------------------------------|
> | <video src='https://user-images.githubusercontent.com/12392404/205430695-505f68cd-0ec8-4525-ab6c-27436339807b.webm' /> | <video src='https://user-images.githubusercontent.com/12392404/205430700-65d6871c-ba82-4f9b-b7af-dff31d21261d.webm' /> | <video src='https://user-images.githubusercontent.com/12392404/205430702-55467701-b813-4624-ad41-882a10684a12.webm'/> |
> 
> Figma link: https://www.figma.com/file/3j9wt6V3lW5jIQSxdW3Eio/ViMusic?node-id=0%3A1&t=D7un5LpecgXNou3f-1
> 
> ##
> 
> >     * I'd like to get rid of ExoPlayer. You know, for better customization and learning purposes.
> This makes sense as long as you wanna learn and explore options. I too wanna learn these stuff in detail for better understanding of the media plaback. But if you are in hurry in releasing a new version of the app then I won't recommend.
> 
> ##
> 
> >     * I'd like to host the playback service into a separate process - so we don't have to worry about requesting battery optimizations. Unfortunately it requires a lot of work (especially due to IPC), so I leave this one for last.
> Recently I have worked on audio playback in seperate service using Exoplayer (can share some insights). It works absolutely fine but I'm not happy the way I've implemented. So definetly need to explore.

Wow, That's impressive! I think option 2 and 3 are better, and the lyrics fonts can be bigger

**Comment by Borttappad** [comment]
> @vfsfitvnm I was busy somewhere else.
> 
> > I still have to figure out where and how to show the current playing track lyrics, stats for nerds and album. That is deciding gestures, animations and whatsoever.
> 
> Here are few ideas for button placements and UI. Refinement is required but yeah this what I have in mind.
> 
> Lyrics style 1	Lyrics style 2	New Button UI
>  Flow.1.1.webm 
>  Flow.2.1.webm 
>  Flow.3.1.webm 
> Figma link: https://www.figma.com/file/3j9wt6V3lW5jIQSxdW3Eio/ViMusic?node-id=0%3A1&t=D7un5LpecgXNou3f-1
> 
> > ```
> > * I'd like to get rid of ExoPlayer. You know, for better customization and learning purposes.
> > ```
> 
> This makes sense as long as you wanna learn and explore options. I too wanna learn these stuff in detail for better understanding of the media plaback. But if you are in hurry in releasing a new version of the app then I won't recommend.
> 
> > ```
> > * I'd like to host the playback service into a separate process - so we don't have to worry about requesting battery optimizations. Unfortunately it requires a lot of work (especially due to IPC), so I leave this one for last.
> > ```
> 
> Recently I have worked on audio playback in seperate service using Exoplayer (can share some insights). It works absolutely fine but I'm not happy the way I've implemented. So definetly need to explore.

I think option 3 is the cleanest looking

**Comment by 825i** [comment]
> For the curious ones, I can only share two screenshots of the new single-screen music player:
> 
> <img alt="" width="25%" src="https://user-images.githubusercontent.com/46219656/203121822-9e364b97-f5db-407b-b85d-0f7f855cbb84.jpg"> <img alt="" width="25%" src="https://user-images.githubusercontent.com/46219656/203121837-9b5d1ee1-0e81-4bf0-97b1-72892dae0815.jpg">

Before I say anything. I realise it's a new app and it's not even an "alpha" look at this stage but boy do I hate it. It's worse than anything I saw in my imagination. Not only do the buttons and UI look like they belong on an Apple device. Everything is tiny, hard to read, has terrible contrast and is going to be absolutely ridiculed by anyone with only half decent vision and regular sized fingers.

I am trying to be nice man, and I hope you see me being facetious but I also hope for everyone's sake that the final product looks absolutely nothing like either of these images. I'm still trying to imagine what the "Now playing" screen is going to look like if I am playing something from eg. (ALL THESE ARE NSFW)

https://en.m.wikipedia.org/wiki/Anarchy_(Chumbawamba_album)

https://en.m.wikipedia.org/wiki/Unfinished_Music_No._1:_Two_Virgins

or the holy grail

https://en.m.wikipedia.org/wiki/No_Love_Deep_Web

Let's be real. Most album art is just shite and don't give me that impressionist "beauty is in the eye if the beholder" bollocks. Gone are the days that we bought CDs for the cover art. That's like buying a book for the cover. It's an afterthought, not just by the listener but also by the musician(s).

As much as we absolutely need new and interesting designs and visionaries (think Aalto). We also shouldn't depart entirely from sanity when it comes to thinking about people who quite frankly aren't as lucky as the rest of us. I'm taking about those not blessed with perfect vision or tiny fingers. Lest you wind up with a fork of your new app called "FMusic - A music app for people with normal sized thumbs". 😂

Anyway man. From one developer to another. I wish you all the best on your journey. If your new app is terrible other people will either fork it or "fix" it. That I'm sure.

**Comment by scottiefalkon** [comment]
What a totally useless post.

**Comment by 825i** [comment]
> I think option 3 is the cleanest looking

Yet it's the only option that steals space from the area which shows song and artist name.

If I had to choose, option 2 is the only one that makes sense. Option 1 doesn't look elegant and option 3 just infringes unnecessarily on another part of the design and it looks too crowded.

**Comment by 825i** [comment]
> What a totally useless post.

I hope you're referring to your own pointless and petty reply.

**Comment by scottiefalkon** [comment]
Petty? You just spilled a bunch of pointless keyboard diarrhea about what
you think something should or should not look like. It's not your project.
No one asked you. And I'm certain no one cares for your smart mouth and
negativity. Grow up.

**Comment by 825i** [comment]
Hahahaha oh my god, what a load of absolute crap. If I ever caught you writing such narrow-minded self righteous pot calling the kettle black horseshit like this and you were in my dev team. I'd have you out of your job faster than you could say "UX".

**Comment by scottiefalkon** [comment]
You'd run your mouth. That's all. That's what people like you do. Add
something positive and constructive or get lost. I've been at this since
94. You're just showing your immaturity.

**Comment by scottiefalkon** [comment]
Didn't read a word. If I wanna hear anything else from you, I'll call on
you. Again, grow up.

**Comment by 825i** [comment]
Nice try. We both know you read every single word. Suck it.

>It's not your project.

God. You really don't know where you are, do you. 🤣
Your whole outburst belongs on #justneckbeardthings

All this from a guy who thinks he's been what, programing for 28 years? What a fraud! 

I've been on Github since what? 2015? 2016? and I've always been able to agree to disagree with someone. No matter how much drama I've seen with hostile takeovers and the like. Never ever seen anything devolve into Twitter-tier nonsense like this and over such a menial post like some UI feedback. 

Thanks for the laugh though. Next time get an enema first or at least try a cup of chamomile before you write something so stupid and forget you aren't on Twitter.

**Comment by scottiefalkon** [comment]
Still typing kid? Feel grown yet?

**Comment by 825i** [comment]
So now I've blocked this low IQ idiot and serial liar (which I admit I should have done before). Now we can get on with the fact that @vfsfitvnm mentioned core UX concepts in his OP.

![Screenshot_20221207_031801_Chrome](https://user-images.githubusercontent.com/58630825/206064251-44f19b17-264b-49c3-9d05-cb5c81fae7a1.jpg)

Personally. Comparing the last 2 screenshots you added @vfsfitvnm. I'm curious is this the "light mode" you're making and will there be a darker mode that doesn't look quite so much like it was made for iOS?

I went back through the original comments you made and collapsed all the others. The more I look at the original redesign. The more I start to like it. Which is why I'm confused you're going with one that I'm sure you know isn't exactly sight impairment friendly.

Could we at least get some kind of comment on the topic of accessibility? There's been plenty around the aesthetic, but what ideas do you have to make everything easy to see and navigate?

**Comment by Kurasami** [comment]
引用:
> Still typing kid? Feel grown yet?

why must you antagonize him? granted, his comment was blunt but this is not a good enough reason to continually provoke him. because this is an open source project he is right when he says anyone can copy the code. so whose project it is does not matter.

there is no rule for how design goes except most people usually focus on function first. the app is made for a purpose and ''''prettiness''' is usually not the main focus of development. usually.

anyway we are all meant to be on the same side, helping to make good software. so putting other ideas down without giving examples is just a bad.

FYI someone has posted this Github issue on another site, I guess after your arguing. so maybe expect more replies from others.

**Comment by lol-lmao-even** [comment]
Couldn't find any evidence that scottiefalkon is employed but I did learn that he smokes weed, is either [lying about being married](https://www.reveddit.com/v/highdeas/comments/wot5u0/deleted_by_user/ikcvs9u/#t1_ikcvs9u) or [his wife is a cuck and he has a bbc fetish](https://old.reddit.com/r/BBCparadise/comments/wz7c4j/she_loves_being_a_bbc_slut_for_her_husband/im3yt22/):

<img width="501" alt="Screenshot 2022-12-07 at 11 18 31" src="https://user-images.githubusercontent.com/120012964/206140552-a24ee57c-dd11-4b2e-9abd-0e2753505f6c.png">

**Comment by senolfndk** [comment]
the new interface is frankly bad it might look like this
![Ekran görüntüsü_20221208_125402](https://user-images.githubusercontent.com/52818836/206417139-969d608f-10da-4320-8ece-b750758b336a.png)

**Comment by vfsfitvnm** [comment]
@rex50 I apologize for the late reply - I was busy working on my custom `AudioPlayer` - I'm not really into UI right now :P.

Regarding the mockup, a progressive blur cannot be implemented, unfortunately. It's just very expensive to compute, so option 2 and 3 are automatically discarded...
Thanks for the mockup by the way - it's impressive you did it having just a single screenshot as source!

Regarding `AudioPlayer`, I have a working (skeleton) code already (the good part is seeking is much faster than `ExoPlayer`).  However, things such as gapless playback, crossfading and silence skipping _look_ challenging to implement, so I don't know if I can make it.

**Comment by Atharv714** [comment]
i think you should make a tutorial to give the people insight of using the app
it will make easy for navigation

**Comment by NeeRaj-2401** [comment]
> tutorial

How u will be paying the dev? 
For this **Tutorial**

**Comment by thelightbringerisme** [comment]
Just do it. Its so good

**Comment by rex50** [comment]
@vfsfitvnm What you are using for AudioPlayer? (as in MediaPlayer or something)

**Comment by vfsfitvnm** [comment]
@rex50 
Neither `MediaPlayer` (which relies on `MediaExtractor`) nor `MediaExtractor` + `MediaCodec` + `AudioTrack` are usable: `MediaExtractor` doesn't support progressive download; i.e. it needs to download the whole track before playing it. However, since API 23, you can pass a custom [`MediaDataSource`](https://developer.android.com/reference/android/media/MediaDataSource#readAt(long,%20byte[],%20int,%20int)) so you _could_ provide your own logic, but again - you can't seek to a specific timestamp without having to download the whole data before it. Yeah, I lost plenty of hours on this.

I eventually shifted to what `ExoPlayer` currently does: custom extractor + `MediaCodec` + `AudioTrack`.

**Comment by NeeRaj-2401** [comment]
**Off-topic**

@vfsfitvnm its already sad to see this project's inactive development since a while because u were working on the new concept. 
And recently you marked this project as archived (m glad its back to normal)

Hope for ur mental as well as physical well-being. :)

**Comment by NeeRaj-2401** [comment]
**Off-topic**

@vfsfitvnm its already sad to see this project's inactive development since a while because u were working on the new concept. 
And recently you marked this project as archived (m glad its back to normal)

Hope for ur mental as well as physical well-being. :)

**Comment by 825i** [comment]
So this was unpinned. Any explanation why? @vfsfitvnm are you done with the redesign or is feedback no longer needed? What's the status on the redesign?

**Comment by vfsfitvnm** [comment]
I had to archive the repository to prevent people from spamming issues (they kept opening duplicate/invalid issues over and over...). I have now limited new accounts from interacting with the repository.

I unpinned this issue because there's no need to keep it pinned anymore - it already received a sufficient amount of feedback.

Regarding the redesign, there is no ETA for it. It could be months, a year, or never.

**Comment by vgovind94** [comment]
Perfect reply.

**Comment by vfsfitvnm** [comment]
Guess what: https://www.theverge.com/2023/3/8/23630821/spotify-design-home-music-podcasts-audiobooks-app

**Comment by ARBoyGo** [comment]
Yeah, I saw that, More like Tiktok/reels/shorts in Music format, What you thinking 🤔 for the new UI 🤣
Well man miss you so much 🙂. 
When is your new app getting released?

**Comment by DarkCrypt** [comment]
Spotify is going to ruin their entire business model? They must be turning retarded. Haha! That's ok, I don't use their app and definetly never will now.

**Comment by Toiletmason** [comment]
If used this feature, and if your trying to listen to music is not that great,  but if you're trying to find new interesting music it's pretty great. It plays like 30 secs of the most played parts of songs so you can gauge whether or not you're going to dig the song or not. That's about the only purpose it serves. A feature like that would have no is in this app is my personal opinion. @vfsfitvnm

**Comment by vfsfitvnm** [comment]
@Toiletmason That's a very interesting idea! I like it - and Youtube (classic) can show the most played parts of a video.

**Comment by Toiletmason** [comment]
I mean, if you could somehow implement a separate tab in the side menu for it I'd probably use it, though I don't save music in vimusic like I do Spotify. It's not tied to an account.

**Comment by Kurasami** [comment]
Jesus I've never seen so much idle chatter in an issues thread before. How the heck didn't I unsubscribe from this.

**Comment by NeeRaj-2401** [comment]
@vfsfitvnm 's prediction / idea was right

<img src="https://github.com/vfsfitvnm/ViMusic/assets/88180887/a342a2be-704d-40d8-bc34-41d579a6bebf" width="30%">


They said :
> “We’re always experimenting with new ways for people to discover music in the YouTube Music app. We’ll consider rolling features out more broadly based on feedback on these experiments.”
>
> GOOGLE SPOKESPERSON

**Comment by ARBoyGo** [comment]
@NeeRaj-2401 You here. Champ

**Comment by TurboSog21** [comment]
too much tiktok-like for me. I don't think everyone would like it if it gets added in. Maybe make it an option in the Appearance/Player tab of the apps settings? Call it something like "Immersive Player" or "Simplefied View" or something similar.
Still looks really good and I'm sure some people would use it instead of the current UI.

**Comment by Kurasami** [comment]
Why the hell am I still getting notifications of replies when I have unsubscribed from this thread.

**Comment by DarkCrypt** [comment]
anandnet must of liked the present ui.

ViMusic(v.0.5.4) basically replicated in Flutter: [Harmony Music](https://github.com/anandnet/Harmony-Music)

**Comment by shafran666** [comment]
I would really love if one could change font size of the lyrics
